lib/rubocop/cop/lint/nested_percent_literal.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for nested percent literals.
+      #
+      # @example
+      #
+      #   # bad
+      #
+      #   # The percent literal for nested_attributes is parsed as four tokens,
+      #   # yielding the array [:name, :content, :"%i[incorrectly", :"nested]"].
+      #   attributes = {
+      #     valid_attributes: %i[name content],
+      #     nested_attributes: %i[name content %i[incorrectly nested]]
+      #   }
+      class NestedPercentLiteral < Cop
+        include PercentLiteral
+
+        MSG = 'Do not nest percent literals.'.freeze
+
+        # The array of regular expressions representing percent literals that,
+        # if found within a percent literal expression, will cause a
+        # NestedPercentLiteral violation to be emitted.
+        REGEXES = PercentLiteral::PERCENT_LITERAL_TYPES.map do |percent_literal|
+          /#{percent_literal}\W/
+        end.freeze
+
+        def on_array(node)
+          process(node, *PercentLiteral::PERCENT_LITERAL_TYPES)
+        end
+
+        def on_percent_literal(node)
+          add_offense(node) if contains_percent_literals?(node)
+        end
+
+        private
+
+        # Returns whether node contains any tokens that match a percent literal
+        # invokation.
+        def contains_percent_literals?(node)
+          node.children.any? do |child|
+            literal = child.children.first.to_s
+            REGEXES.any? { |regex| literal.match? regex }
```

Comment:
We support running RuboCop on old rubies (2.0.0 and above) which means you can't use newer features like `#match?` in the codebase. 🙂

Reply:
Yep, I see from the Travis CI failures. Updating to use `String#match` rather than `String#match?`.